test/e2e/util_test.go

```diff
@@ -259,6 +260,37 @@ func buildRoute(name, namespace, serviceName string) *routev1.Route {
 	}
 }
 
+// buildRouteWithHTTPHeaders returns a route definition targeting the specified service along with setting of header name and value.
+func buildRouteWithHTTPHeaders(name, namespace, serviceName string, headerName string, headerValue string) *routev1.Route {
```

Comment:
This should have a parameter for Action and a switch on Action so you can test Delete as well.